router/throttler/throttler.go

```diff
@@ -47,18 +50,30 @@ func (f *Factory) Get(destName, destID string) *Throttler {
 	f.throttlersMu.Lock()
 	defer f.throttlersMu.Unlock()
 	if t, ok := f.throttlers[destID]; ok {
+		if f.adaptiveRateLimiter != nil {
+			t.config.limit = f.adaptiveRateLimiter(destName, destID, t.config.limit)
+		}
 		return t
 	}
 
 	var conf throttlingConfig
 	conf.readThrottlingConfig(destName, destID)
+	if f.adaptiveRateLimiter != nil {
+		conf.limit = f.adaptiveRateLimiter(destName, destID, conf.limit)
+	}
 	f.throttlers[destID] = &Throttler{
 		limiter: f.limiter,
 		config:  conf,
 	}
 	return f.throttlers[destID]
 }
 
+func (f *Factory) SetLimitReached(destID string) {
+	f.throttlersMu.Lock()
+	defer f.throttlersMu.Unlock()
+	f.limitReachedPerDestination <- destID
+}
+
```

Comment:
I have a similar comment in  `adaptiveRatelimit.go`. 

```suggestion
func (f *Factory) SetLimitReached(destID string) {
	f.throttlersMu.Lock()
	defer f.throttlersMu.Unlock()
	f.limitReachedPerDestination.limitReached(destID)
}
```